DittoHeartbeat/src/main/java/live.ditto.dittoheartbeat/Heartbeat.kt

```diff
@@ -0,0 +1,149 @@
+package live.ditto.dittoheartbeat
+
+import live.ditto.Ditto
+import android.os.Handler
+import android.os.Looper
+import androidx.compose.material3.Text
+import live.ditto.DittoConnectionType
+import live.ditto.DittoPeer
+import live.ditto.DittoPresenceObserver
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+data class HeartbeatConfig(
+    val interval: Long,
+    val collectionName: String,
+    val metaData: Map<String, Any>
```

Comment:
What do you think about using a data class here instead of a map?

```
data class Metadata(
    val deviceId: String,
    val locationId: String,
    ...
)
```

With a data class in place we can make use of extension functions to convert it to a map.

```
fun Metadata.toMap() = mapOf(
    "deviceId": deviceId,
    "locationId": locationId,
    ...
)
```

Reply:
The metadata was initially intended for passing in the id's (ex. locationId and deviceId) that would then create the composite id in the ditto collection. I've since updated the config to have an id key where the user will pass all of the id's they want to use to create their composite key. So it leaves me with not knowing what the metadata would be used for (in the current state of the heartbeat tool) so I might remove it til its needed.